locator/token_metadata.cc

```diff
@@ -64,6 +64,8 @@ class token_metadata_impl final {
 
     std::unordered_map<token, inet_address> _bootstrap_tokens;
     std::unordered_set<inet_address> _leaving_endpoints;
+    // The map between the existing node to be replaced and the replacing node
+    std::unordered_map<inet_address, inet_address> _replacing_endpoints;
```

Comment:
We will when we support replacing nodes in parallel. The work is in progress.

Reply:
As long as it will not create a situation where more than one pending node exists for a range it should be fine I guess.

Reply:
It should not. Because the replaced nodes are supposed to have different ranges.

Reply:
A range becomes pending not only for primary replicas, but secondaries as well. IOW if you replace one node in 3 node cluster all ranges will have pending node and if you will try to replace two they will have two pending nodes which will not work. I think we are too far from been able to support more that one pending node. LWT outright check for that and throws an error. So lest code for one replacing node (at least per DC) for now.

Reply:
Yes, it will take a while to support replacing with multiple nodes. It is already coded as supporting multiple replacing node in token_metadata. It is not much more complicated than supporting a single node.  Btw, We have this multiple nodes support in token_metadata for bootstrap too, std::unordered_map<token, inet_address> _bootstrap_tokens.